src/Validation/Rules/DateTimeFormat.php

```diff
@@ -6,17 +6,23 @@
 
 use Attribute;
 use DateTime;
+use DateTimeInterface;
 use Tempest\Validation\Rule;
 
 #[Attribute]
 final readonly class DateTimeFormat implements Rule
 {
-    public function __construct(private string $format = 'Y-m-d')
+    public function __construct(public string $format = 'Y-m-d')
     {
     }
 
     public function isValid(mixed $value): bool
     {
+        $value = match ($value instanceof DateTimeInterface) {
```

Comment:
In general I prefer to write it like this:

```php
match ($value::class)
{
    DateTimeInterface::class => …
    default =>
```

That's because PhpStorm then knows which class it's dealing with within the match arms.

Reply:
We can't do $value::class here because we are unsure if it's an object, we'd need an additional check

Reply:
Good point!